teuthology/packaging.py

```diff
@@ -970,6 +970,39 @@ def repo_url(self):
             'repo',
         )
 
+    @property
+    def build_complete(self):
+        # use the repo search results to get a ref and a sha1; the
+        # input to teuthology-suite doesn't contain both
+        try:
+            self.assert_result()
+        except VersionNotFoundError:
+            return False
+
+        search_result = self._result.json()[0]
+
+        # now look for the build complete status
+        build_url = '/'.join((
+            self.query_url.rstrip('/'),
+            'builds/ceph',
+            search_result['ref'],
+            search_result['sha1'],
+        ))
```

Comment:
Check out `urljoin` - it's already imported here!

Reply:
Yeah...it only seems to do two at a time, which seemed annoying.  If it's still better I can use it.

Reply:
Compromise to hopefully get maximum benefit with least repetition.